### DISCUSSION TITLE
Generics will not be auto-imported

### DISCUSSION THREAD

**Discussion by williamfzc** [square/javapoet#921] [General]
Hi~
I used javapoet for guessing some generic types.

```
Log.info("guess: " + guessed);
methodBuilder.addStatement("$T $N", guessed, eachParam.getName());
```

console:

```
guess: List<Long>
```

But it did not import its class correctly:

![image](https://user-images.githubusercontent.com/13421694/177968254-c5e0cbdd-55c4-4c16-96d7-2392a988414e.png)

any suggestions? 
Thanks for your help!

**Comment by JakeWharton** [comment]
What is `guessed`? It should be a `ParameterizedTypeName` of a `ClassName` of `java.util.List` and whose parameter is a `ClassName` of `java.util.Long`.

**Reply by williamfzc** [reply]
![image](https://user-images.githubusercontent.com/13421694/178397297-eea6bc09-7362-41ac-b908-a95cc20aea36.png)

typeForGuess is a string `List<Long>`

**Reply by JakeWharton** [reply]
Parsing a string like `"List<Long>"` is going to be very complicated. You need to consider things like the default import set and the parser needs to be recursive for nested parameters.

**Comment by williamfzc** [comment]
Should I:
- split it into 2 parts, List and Long
- get ClassName of List and Long
- combine 2 ClassName into `guessed`

right?
